js/bundle_test.go

```diff
@@ -373,65 +405,54 @@ func TestNewBundle(t *testing.T) {
 	})
 }
 
-func TestNewBundleFromArchive(t *testing.T) {
-	fs := afero.NewMemMapFs()
-	assert.NoError(t, fs.MkdirAll("/path/to", 0755))
-	assert.NoError(t, afero.WriteFile(fs, "/path/to/file.txt", []byte(`hi`), 0644))
-	assert.NoError(t, afero.WriteFile(fs, "/path/to/exclaim.js", []byte(`export default function(s) { return s + "!" };`), 0644))
-
-	data := `
-			import exclaim from "./exclaim.js";
-			export let options = { vus: 12345 };
-			export let file = open("./file.txt");
-			export default function() { return exclaim(file); };
-		`
-	b, err := getSimpleBundleWithFs("/path/to/script.js", data, fs)
-	if !assert.NoError(t, err) {
-		return
-	}
-	assert.Equal(t, lib.Options{VUs: null.IntFrom(12345)}, b.Options)
-
-	bi, err := b.Instantiate()
-	if !assert.NoError(t, err) {
-		return
-	}
-	v, err := bi.Default(goja.Undefined())
-	if !assert.NoError(t, err) {
-		return
+func getArchive(data string, rtOpts lib.RuntimeOptions) (*lib.Archive, error) {
+	b, err := getSimpleBundle("script.js", data, rtOpts)
+	if err != nil {
+		return nil, err
 	}
-	assert.Equal(t, "hi!", v.Export())
-
-	arc := b.makeArchive()
-	assert.Equal(t, "js", arc.Type)
-	assert.Equal(t, lib.Options{VUs: null.IntFrom(12345)}, arc.Options)
-	assert.Equal(t, "file:///path/to/script.js", arc.FilenameURL.String())
-	assert.Equal(t, data, string(arc.Data))
-	assert.Equal(t, "file:///path/to/", arc.PwdURL.String())
+	return b.makeArchive(), nil
+}
 
-	exclaimData, err := afero.ReadFile(arc.Filesystems["file"], "/path/to/exclaim.js")
-	assert.NoError(t, err)
-	assert.Equal(t, `export default function(s) { return s + "!" };`, string(exclaimData))
+func TestNewBundleFromArchive(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		testCases := []struct {
+			cm   compiler.CompatibilityMode
+			code string
+		}{
+			{compiler.CompatibilityModeExtended,
+				`export let options = { vus: 12345 };
+				export default function() { return "hi!"; };`},
+			{compiler.CompatibilityModeBase,
+				`module.exports.options = { vus: 12345 };
+					module.exports.default = function() { return "hi!" };`},
+		}
```

Comment:
nit: at least in github the js code looks not well formatted ;)

Reply:
You mean the misalignment? Yeah, it's not great. I wanted to avoid the initial newline used in other tests, but legibility should be more important.

I'll fix it.

Reply:
Fixed in 45407364.